src/app/loginComponents/state/requests.service.ts

```diff
@@ -186,4 +189,26 @@ export class RequestsService {
         }
       );
   }
+
+  deleteOrganization(id: number, isAdminOrCurator: boolean): void {
+    this.alertService.start('Deleting organization ' + id);
+    this.requestsStore.setLoading(true);
+    this.openApiOrgService
+      .deleteRejectedOrPendingOrganization(id)
+      .pipe(finalize(() => this.requestsStore.setLoading(false)))
+      .subscribe(
+        (organization: Organization) => {
+          this.alertService.simpleSuccess();
+          if (isAdminOrCurator) {
+            this.updateCuratorOrganizations();
+          }
+          this.updateMyMemberships();
+        },
+        () => {
+          this.updateMyMembershipState(null, null, null, null);
```

Comment:
What is this doing? Presumably the state hasn't changed since the delete failed. Is there any state that was changed that you need to rollback here?

Reply:
Just chatted about this. That line is unnecessary since there's really no point in resetting the state on a fail case if the state has not been changed. However, leaving this line in as it fits in with the patterns on lines 121, 167, and 186.